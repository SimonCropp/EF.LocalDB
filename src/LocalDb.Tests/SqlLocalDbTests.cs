using System.Diagnostics;
using Xunit;
using Xunit.Abstractions;

public class SqlLocalDbTests :
    XunitLoggingBase
{
    [Fact]
    public void Instances()
    {
        var collection = LocalDbApi.GetInstanceNames();
        foreach (var instance in collection)
        {
            Trace.WriteLine(instance);
        }
        Assert.NotEmpty(collection);
    }

    [Fact]
    public void NonInstanceInfo()
    {
        var info = LocalDbApi.GetInstance("Missing");
        Assert.False(info.Exists);
    }

    [Fact]
    public void Info()
    {
        LocalDbApi.CreateAndStart("InfoTest");
        var info = LocalDbApi.GetInstance("InfoTest");

        ObjectApprover.Verify(info);
        LocalDbApi.StopAndDelete("InfoTest");
        Assert.False(LocalDbApi.GetInstance("InfoTest").Exists);
    }

    //[Fact]
    //public void DeleteAll()
    //{
    //    foreach (var instance in LocalDbApi.GetInstanceNames())
    //    {
    //        Trace.WriteLine(instance);
    //        LocalDbApi.StopAndDelete(instance);
    //    }
    //}

    public SqlLocalDbTests(ITestOutputHelper output) :
        base(output)
    {
    }
}