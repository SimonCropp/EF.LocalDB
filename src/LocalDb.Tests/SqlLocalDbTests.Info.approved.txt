{
  Size: 1460,
  InstanceName: 'InfoTest',
  Exists: true,
  IsRunning: true,
  SharedInstanceName: ''
}